mmtrack/models/mot/qdtrack.py

```diff
@@ -0,0 +1,197 @@
+            proposal_cfg = self.detector.train_cfg.get(
+                'rpn_proposal', self.detector.test_cfg.rpn)
+            rpn_losses, proposal_list = self.detector.rpn_head.forward_train(
+                x,
+                img_metas,
+                gt_bboxes,
+                gt_labels=None,
+                gt_bboxes_ignore=gt_bboxes_ignore,
+                proposal_cfg=proposal_cfg)
+            losses.update(rpn_losses)
+
+        ref_x = self.detector.extract_feat(ref_img)
+        ref_proposals = self.detector.rpn_head.simple_test_rpn(
+            ref_x, ref_img_metas)
+
+        roi_losses = self.detector.roi_head.forward_train(
+            x, img_metas, proposal_list, gt_bboxes, gt_labels,
+            gt_bboxes_ignore, gt_masks)
+
+        losses.update(roi_losses)
+
+        track_losses = self.track_head.forward_train(
+            x, img_metas, proposal_list, gt_bboxes, gt_labels,
+            gt_match_indices, ref_x, ref_img_metas, ref_proposals,
+            ref_gt_bboxes, ref_gt_labels, gt_bboxes_ignore, gt_masks,
+            ref_gt_bboxes_ignore, **kwargs)
+
+        assert track_losses
+
+        losses.update(track_losses)
+
+        return losses
+
+    def simple_test(self, img, img_metas, rescale=False):
+        # TODO inherit from a base tracker
+        assert self.track_head.with_embed_head, 'Embed head must be implemented.'  # noqa
+        frame_id = img_metas[0].get('frame_id', -1)
+
+        x = self.detector.extract_feat(img)
+        proposal_list = self.detector.rpn_head.simple_test_rpn(x, img_metas)
+
+        det_bboxes, det_labels = self.detector.roi_head.simple_test_bboxes(
+            x,
+            img_metas,
+            proposal_list,
+            self.detector.test_cfg.rcnn,
+            rescale=rescale)
+
+        det_bboxes = det_bboxes[0]
+        det_labels = det_labels[0]
+
+        if det_bboxes.size(0) == 0:
+            track_feats = None
+        else:
+            track_bboxes = det_bboxes[:, :-1] * torch.tensor(
+                img_metas[0]['scale_factor']).to(det_bboxes.device)
+            track_feats = self.track_head._track_forward(x, [track_bboxes])
+
+        if track_feats is not None:
+            bboxes, labels, ids = self.tracker.track(
+                bboxes=det_bboxes,
+                labels=det_labels,
+                track_feats=track_feats,
+                frame_id=frame_id)
+
+        det_bboxes = bbox2result(det_bboxes, det_labels,
+                                 self.detector.roi_head.bbox_head.num_classes)
+
+        if track_feats is not None:
+            track_bboxes = outs2results(
+                bboxes=bboxes,
+                labels=labels,
+                ids=ids,
+                num_classes=self.detector.roi_head.bbox_head.num_classes
+            )['bbox_results']
+        else:
+            track_bboxes = [
+                np.zeros((0, 6), dtype=np.float32)
+                for _ in range(self.detector.roi_head.bbox_head.num_classes)
```

Comment:
do not use `self.detector.roi_head.bbox_head.num_classes` to get the number for classes.
you can use `len(det_bboxes)` in line 117